T6_1/JSONRepository.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Text.Json;
+using System.Configuration;
+using Homewokr_6_1;
+
+namespace Homework_6_1
+{
+	public class JSONRepository : IRepository<Catalog>
+	{
+		protected virtual string DirectoryPath { get; set; } = "json_directory";
+
+		public JSONRepository()
+		{
+			DirectoryPath = ConfigurationManager.AppSettings["Repositories"] ?? DirectoryPath;
+		}
+
+		public void SaveCatalog(Catalog catalog, string filePath)
+		{
+			if (!Directory.Exists(filePath))
+			{
+				Directory.CreateDirectory(filePath);
+			}
+
+			var authorGroups = catalog.SelectMany(book => book.Authors.Select(author => (author, book)))
+									   .GroupBy(tuple => tuple.author);
+
+			foreach (var group in authorGroups)
+			{
+				var author = group.Key;
+				string authorFileName = $"{author.FirstName}_{author.LastName}.json";
+				string fullFilePath = Path.Combine(filePath, authorFileName);
```

Comment:
Not bad solution to generate class to store book + isbn on the stage of saving, and use dynamic during reading, but it can be dangerous (no type check on compile time). Can use this approach, but keep in mind the possibility to create class for Book on DAL layer, and use it for serialize/deserialize operations. It will be more reliable and no need to read properties in JSON directly (GetProperty() can throw exception which has to be proceed and not block the reading of other entities from repo).